reconcile/aus/base.py

```diff
@@ -739,6 +784,49 @@ def verify_lock_should_skip(
     return False
 
 
+def _create_upgrade_policy(
+    next_schedule: str, spec: ClusterUpgradeSpec, version: str
+) -> AbstractUpgradePolicy:
+    if spec.cluster.is_rosa_hypershift():
+        return ControlPlaneUpgradePolicy(
+            cluster=spec.cluster,
+            version=version,
+            schedule_type="manual",
+            next_run=next_schedule,
+        )
+    return ClusterUpgradePolicy(
+        cluster=spec.cluster,
+        version=version,
+        schedule_type="manual",
+        next_run=next_schedule,
+    )
+
+
+def _calulate_node_pool_diffs(
+    ocm_api: OCMBaseClient, spec: ClusterUpgradeSpec, now: datetime
+) -> Optional[UpgradePolicyHandler]:
+    node_pools = get_node_pools(ocm_api, spec.cluster.id)
+    if node_pools:
+        for pool in sorted(node_pools, key=lambda x: x["id"]):
```

Comment:
Why sort here? It seems we only want to return at most one upgrade policy, is that expected? Why not upgrade all mismatch ones?

Reply:
Right, this can actually go away now, this was from when I was not using mutexes. Will remove it.